wordpress-mobile/WordPress-Android
# Migrate quick start notification reminders

Parent #15136 

This PR reintroduces reminder notifications into the Improved My Site flow.

**To test:**
- Launch the app 
- Navigate to My Site -> App Settings -> Test feature configuration
- Enable the `MySiteImprovementsFeatureConfig` and tap the button to restart the app (I also like to swipe the app closed after that and then reopen - sometimes weird behavior happens)
- Create a new site
- When the Quick Start Prompt is shown, tap the "Show me around" option
- Tap the `Customize your site` row to launch the QS dialog
- Tap on `Check your site title` (the dialog will close and you'll be brought back to My Site)
- Tap on the blue flashing icon on site title and change the text
***Notifications are set for two days in the future, so you have to change the date/time on the device for testing***
- Navigate out of the app to the Settings area of your device and locate the area that manages the time.
- Disable the "use network-provided time" 
- Manually set the date/time for 2 days and 2 minutes from now
- NOTE: The notification should arrive at the device within a couple of minutes
- After the notification is shown - **don't forget to enable "use network-provided time"**


## Regression Notes
1. Potential unintended areas of impact
Quick start doesn't show reminders or notifications

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tests

3. What automated tests I added (or what prevented me from doing so)
New QuickStartRepositoryTest tests for verifying kickoff of notification/reminder method

PR submission checklist:
